First Code Review

nadiai opened the pull request:
I know most of this isn't technically correct but I have a basic outline of what I would like it to look like and I'd love to expand it correctly.

nadiai committed:
Initial Commit

I just copied my file over from the static folder.

nadiai committed:
this is not my second commit

the actual second commit failed.. I'm upset because I dont know what happened to my work.. I was happy. happiness doesn't last too long I see.

nadiai committed:
Third commit

I made some revisions.

nadiai committed:
5/21 Commit 1

Yayyy. I've made actual progress. I had to work around some software issues, but this is stuff I was working on on Friday.

nadiai committed:
5/21 2nd commit

More revisions.

nadiai committed:
5/21 Final work commit

I'm trying to figure some different styles to implement.

nadiai committed:
5/22 Commit

Learned that I was spending too much time and little things so I had to move on even though it's incomplete.

nadiai committed:
5/23 Commit at work

I worked on php a lot today, converted some files to php, but continued to edit them in html. I'm not finished but will soon do a code review to spare myself heartache in the future.

nadiai committed:
5/24 First Commit

I have tentatively finished my website and I'm ready for a preliminary code review.

nadiai committed:
5/24 Final at work Commit

I worked in php some more!

tilikang commented on Blogpost1.php:
```diff
@@ -0,0 +1,28 @@
+
+		<link href='https://fonts.googleapis.com/css?family=Barrio' rel='stylesheet'>
+
+		<link href='https://fonts.googleapis.com/css?family=Delius Unicase' rel='stylesheet'>
```
It should be possible to include both of the fonts in a single `<link>` tag instead of using two. This will make your site a bit faster because it'll only need to make one request to Google instead of two.

tilikang commented on Blogpost1.php:
```diff
@@ -0,0 +1,28 @@
+	</head>
+	<body class="backdrop">
+		<h1> My Interests</h1>
+		<img class="test" src="/beyhive.jpg" alt="Beyhive">
```
This is fine, but as your code base gets bigger, you'll want to organize your files a bit. Maybe make a folder called `/images` where you can put all the images so they don't get mixed up with your html/php files

tilikang commented on Inventory.spf:
```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
What's going on with this file?

nadiai replied:
I have no idea when/how/where/if i created actually

tilikang commented on Practice in php/calculator.php:
```diff
@@ -0,0 +1,29 @@
+<?php
```
Just a general comment on some of these file/folder names: I'd suggest not using spaces in any folders or files. Spaces aren't technically valid characters in a website address, so while it might work, it's not guaranteed to. Instead, try using underscores ("_") to replace the spaces.

tilikang commented on Practice in php/colorcalculator.php:
```diff
@@ -0,0 +1,42 @@
+
+if(isset($_REQUEST['ColorSelector1']) && $_REQUEST['ColorSelector1']=='Red' || isset($_REQUEST['ColorSelector2']) && $_REQUEST['ColorSelector2']=='Red'){
+		echo '
+		<body style="background-color:#af2323">';
```
Rather than echoing the full body tag each time, you could save a variable called `$backgroundColor` or something like that, and then below all of these if statements say:
```
<body style='background-color:$backgroundColor;'>
```

tilikang commented on Practice in php/colorcalculator.php:
```diff
@@ -0,0 +1,42 @@
+<?php
+
+
+if(isset($_REQUEST['ColorSelector1']) && $_REQUEST['ColorSelector1']=='Red' || isset($_REQUEST['ColorSelector2']) && $_REQUEST['ColorSelector2']=='Red'){
```
It looks like this wasn't totally finished. That's fine, but let me know if you want to talk through how you could handle all the logic in these if statements.

tilikang commented on Resume.html:
```diff
@@ -0,0 +1,37 @@
+		<Body style='background-image:url(/wheat.jpg)'>
+			<h1> Resume</h1>
+			<table>
+				<th class='Header'> Education</th>
```
Something you could do to simplify this just a bit: instead of giving a class to all the `<th>` elements in this table, give a class to the table itself (e.g. "resumeTable"), and then do this in css:
```
.resumeTable th{
   /*put your style here*/
}
```
That CSS is saying "find all the <th> tags inside the resumeTable.

tilikang commented on Resume.html:
```diff
@@ -0,0 +1,37 @@
+				<tr>
+					<td> Olin Business School at Washington University in St. Louis Class of 2020</td>
+				</tr>
+				<th class='Header'> Activities</th>
```
`<th>` elements are like `<td>`s. They should be inside the `<tr>`

tilikang commented on Style.css:
```diff
@@ -0,0 +1,62 @@
+	border-bottom-color:#0a2e68;
+}
+
+table{
```
It looks to me like this style is meant to apply to a specific table, but you're applying it to *all* tables. That might work right now, but it makes the code hard to maintain because you can't add any other tables later. Instead, give this specific table a class and use that to apply CSS to it.

tilikang commented on blogpost.css:
```diff
@@ -0,0 +1,98 @@
+}
+
+h1{
+			font-family: 'Barrio';
```
There's some funky indentation going on here.

tilikang commented on blogpost.css:
```diff
@@ -0,0 +1,98 @@
+	text-align: center;
+	background-color: #fff;
+	border: 8px solid #fcce16;
+	-webkit-border-radius: 30px;
```
these "prefix" things aren't really necessary anymore. Back in the day we needed to use them, but just `border-radius` should be fine now.

tilikang reviewed (commented):
Nice! Your html is really clean and the css looks like you've done quite a bit of experimentation. I left some comments, but let me know if you have any other questions.

Whenever you're ready, you can try making your blog dynamic (let me know if you want to talk about that)

nadiai committed:
5/28 Commit

I finalized a second draft of my blog and im ready to move on to the php conversion

nadiai committed:
5/29 Commit

I spent most of my time doing and undoing one thing.. much work to be done!

nadiai committed:
5/30 Mid-day Commit

Khaila guided me through creating forms!

nadiai committed:
06/01 Commit

I tried to implement MySQL and  PHP

nadiai committed:
06/04 Midday commit

Yayyyy organization! I wokred on organizing my work most of the day! it was tedious but gratifying

nadiai committed:
06/04 2nd commit

Maintaining style across all pages

nadiai committed:
06/05 First commit

I finally implemented all the changes from the last code review and I think I'm ready for my second.

nadiai committed:
06/05 Second commit

I like to go through the github version of my work cause it looks so different from the atom version. So I made some stylistic changes and I'm now ready for the code review. I am yet to go back and change the color calculator because I just wanted to keep moving but I will go back and correct it.

nadiai committed:
6/05 third commit

Before switching branches so nothigns final

nadiai committed:
06/07 Commit

YAYYYY! I made comments show up!

nadiai commented:
Hey Tyler could you review functions.php, blogpost.php, header.php (i recognize that i need to rename it and create an actual header), and websiteMasterStyle.css.

tilikang commented on Blog/include/functions.php:
```diff
@@ -0,0 +1,77 @@
+
+}
+
+$allBlogPosts = GetAllBlogPosts();
```
include files like this should define functions, but they should only be called from inside other functions, or from the main web page (i.e. files not in the include folder). Right now, you're calling `GetAllBlogPosts` on every page load, even when you don't need it.

tilikang commented on Blog/include/functions.php:
```diff
@@ -0,0 +1,77 @@
+	return $result;
+}
+
+$allComments = getAllComments();
```
Same thing here. This should be removed and called from the web page where you actually need this data

tilikang commented on Blog/include/header.php:
```diff
@@ -0,0 +1,45 @@
+<html>
```
When you have re-usable html like this in the include folder, you want to wrap it in a function, and call that function from the actual web page. That way you can include this file in your `includeAll.php` file, and then at the top of any page where you want the header printed, just call `PrintHeader` or whatever you decide to call the function.

tilikang commented on Blog/include/includeAll.php:
```diff
@@ -0,0 +1,6 @@
+
+include('Blog/include/functions.php');
+//include('Blog/include/websiteMasterStyle.css')
+echo "<link href='Blog/include/websiteMasterStyle.css' rel='stylesheet'>";
```
This shouldn't go here. If you look at the html of a page on your website, you'll see that this link tag is at the very top, outside of the `<html>` tag. generally speaking, you don't want to echo anything automatically like this (all echos should be inside a function you have to call) because that prevents you from running any code prior to this being echoed. Once something is echoed, you can't redirect the page somewhere else, so you can't really process forms properly with this approach. Let me know if you want to discuss this more, but the general rule is, only echo stuff: (1) inside a function, or (2) from an actual web page, not an include file.

In this case, just throw this link in the `<head>` tag in header.php.

tilikang commented on Blog/include/websiteMasterStyle.css:
```diff
@@ -0,0 +1,358 @@
+	 top: 100px;
+	 border: 15px solid;
+	 border-color: #fff transparent transparent #fff;
+}
```
It's hard to tell if CSS is working just from reading the code, but nothing seems off to me. If it looks the way you want it to, then this all seems good.

tilikang reviewed (commented):
@nadiai - I'm not able to find a file called blogpost.php in this PR. Is it called something else? Or maybe it didn't get committed yet?

Aside from that, things are looking good. I left a few comments about putting things in functions, but nothing major. Once you're done with this PR, I'd suggest merging it into master, and going forward, make a new branch for each new feature/project, and PR those into staging.
